File: packages/node-sdk/src/utils/retry.ts
export function exponentialBackoff(attempt: number, baseDelay: number = 1000): number {
  const maxDelay = 30000; // 30 seconds max
  const delay = Math.min(baseDelay * Math.pow(2, attempt - 1), maxDelay);
  
  // Add jitter (±20%)
  const jitter = delay * 0.2 * (Math.random() - 0.5);
  
  return Math.floor(delay + jitter);
}

File: packages/python-sdk/powerlobster_webhook/utils/retry.py
"""Retry utilities with exponential backoff."""

import random


def exponential_backoff(attempt: int, base_delay: float = 1.0) -> float:
    """
    Calculate exponential backoff delay with jitter.
    
    Args:
        attempt: Retry attempt number (1-indexed)
        base_delay: Base delay in seconds
    
    Returns:
        Delay in seconds
    """
    max_delay = 30.0  # 30 seconds max
    delay = min(base_delay * (2 ** (attempt - 1)), max_delay)
    
    # Add jitter (±20%)
    jitter = delay * 0.2 * (random.random() - 0.5)
    
    return delay + jitter
